### ISSUE TITLE
How could I add extra pre-processing operations only to the input of Discriminator? 

### ISSUE THREAD

**Issue by HJLYU0519** [junyanz/CycleGAN#110]
I made some changes to the Discriminator. And I want to add some extra pre-processing operations only to the input of Discriminator. What should I change? 
And I think there are no preprocessing operations. my reason is this:  in data/__init__.py 
         75  self.dataloader = torch.utils.data.DataLoader(
         76     self.dataset,
         77     batch_size=opt.batch_size,
         78     shuffle=not opt.serial_batches,
         79     num_workers=int(opt.num_threads))

Should I establish another dataloader only for Discriminator?
Many thanks for your kind help

**Comment by junyanz** [comment]
That is one possibility. You can also manually add some preprocessing code before you call the netD.forward. Like [here](https://github.com/junyanz/CycleGAN/blob/master/models/cycle_gan_model.lua#L141).
